compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETileAndFuse.cpp

```diff
@@ -288,6 +321,20 @@ LogicalResult applyTileAndFuse(RewriterBase &rewriter, Operation *rootOp,
   return success();
 }
 
+LogicalResult applyTileAndFuseUsingSCF(RewriterBase &rewriter,
+                                       Operation *rootOp,
+                                       DominanceInfo &dominanceInfo,
+                                       scf::SCFTilingOptions options,
+                                       int64_t tilingLevel) {
+  if (tilingLevel == 3) {
+    return applyTileAndFuseUsingSCFFor(rewriter, rootOp, dominanceInfo,
+                                       options);
+  } else {
```

Comment:
Can we separate TileAndFuseUsingSCFFor without using tilingLevel as 3? One reason to do so is in packing pipeline, we have tiling organized as scf.forall - scf.for - scf.forall. It would be weird to use tilingLevel 1, 3 and 2...

Reply:
I'm not sure if I understood the requirement (and thus the solution) properly but adding a few points to consider. Let me know your thoughts.

I guess as a temporary workaround we should indeed club it as 1 -> 3 -> 2, for now since all these are in their very initial stages.

I can do the following :-
```
Pack =>
tileFuse(1)
tileFuse()
tileFuse(2)

Pad =>
tileFuse(1)
tileFuse(2)
tileFuse()
```
The above would be one way to avert the numbering weirdness - but it's just not tackling the real issue and is still a temporary workaround.

I'd therefore still suggest to go ahead with the current PR because of the following reasons :-
1. @nirvedhmeshram and I are implementing set/get functionality for `lowering_config` + `translation_info` - this would help in having a separate pipeline for pad and pack based approaches along with the expected tiling sizes at each level. One of the constraints for the pack based approach should be addressed by this.
2. @MaheshRavishankar mentioned that with his upstream PR we would be able to better control whether to use scf.for or scf.forall and also club them with tiling level while invoking the pass itself as flag/template - and these constructs would soon evolve in a better shape iteratively. The other constraint of clubbing 1->2->3 or 1->3->2 should be addressed by this incrementally.
3. Since this is a first complete e2e C++ pipeline - any other implementation/fixes we make would be on top of this so we should at least have this revision (working for pad based flow) in `main`.